main.go

```diff
@@ -33,28 +33,66 @@ func (a unencryptedAuth) Start(server *smtp.ServerInfo) (string, []byte, error)
 	return a.Auth.Start(&s)
 }
 
-func main() {
-	cfg := config.Get()
+// App version informaton retrieved on runtime
+var (
+	// BuildTime represents the time in which the service was built
+	BuildTime string
+	// GitCommit represents the commit (SHA-1) hash of the service that is running
+	GitCommit string
+	// Version represents the version of the service that is running
+	Version string
+)
 
+func main() {
 	log.Namespace = "dp-frontend-dataset-controller"
 
+	signals := make(chan os.Signal, 1)
+	signal.Notify(signals, os.Interrupt, os.Kill)
+
+	ctx := context.Background()
+
+	cfg, err := config.Get()
+	if err != nil {
+		log.Event(ctx, "unable to retrieve service configuration", log.Error(err))
+		os.Exit(1)
+	}
+
+	log.Event(ctx, "got service configuration", log.Data{"config": cfg})
+
+	versionInfo, err := health.NewVersionInfo(
+		BuildTime,
+		GitCommit,
+		Version,
+	)
+	if err != nil {
+		log.Event(ctx, "failed to create service version information", log.Error(err))
+		os.Exit(1)
+	}
+
 	router := mux.NewRouter()
 
 	f := filter.New(cfg.FilterAPIURL)
-	zc := zebedee.NewZebedeeClient(cfg.ZebedeeURL)
+	zc := zebedee.New(cfg.ZebedeeURL)
 	dc := dataset.NewAPIClient(cfg.DatasetAPIURL)
 	rend := renderer.New(cfg.RendererURL)
 
-	router.StrictSlash(true).Path("/healthcheck").HandlerFunc(healthcheck.Handler)
+	healthcheck := health.New(versionInfo, cfg.HealthCheckCriticalTimeout, cfg.HealthCheckInterval)
+
+	if err = registerCheckers(ctx, &healthcheck, f, zc, dc, rend); err != nil {
+		log.Event(ctx, "failed to add checkers", log.Error(err))
```

Comment:
Maybe we can remove this log.Event? (We are already logging what checkers failed to register in the registerCheckers function, and I think the error at this point only be the last one if there was more than one)